rmf_demos_tasks/rmf_demos_tasks/dispatch_delivery.py

```diff
@@ -50,60 +61,99 @@ def __init__(self, argv=sys.argv):
-
-    def generate_task_req_msg(self):
-        req_msg = SubmitTask.Request()
-        req_msg.description.task_type.type = TaskType.TYPE_DELIVERY
-
-        delivery = Delivery()
-        delivery.pickup_place_name = self.args.pickup
-        delivery.pickup_dispenser = self.args.pickup_dispenser
-        delivery.dropoff_place_name = self.args.dropoff
-        delivery.dropoff_ingestor = self.args.dropoff_ingestor
-        req_msg.description.delivery = delivery
-
-        ros_start_time = self.node.get_clock().now().to_msg()
-        ros_start_time.sec += self.args.start_time
-        req_msg.description.start_time = ros_start_time
-
-        req_msg.description.priority.value = self.args.priority
-        return req_msg
-
-    def main(self):
-        if not self.submit_task_srv.wait_for_service(timeout_sec=3.0):
-            self.node.get_logger().error('Dispatcher Node is not available')
-            return
-
-        rclpy.spin_once(self.node, timeout_sec=1.0)
-        req_msg = self.generate_task_req_msg()
-        print(f"\nGenerated delivery request: \n {req_msg}\n")
-        self.node.get_logger().info("Submitting Delivery Request")
-
-        try:
-            future = self.submit_task_srv.call_async(req_msg)
-            rclpy.spin_until_future_complete(
-                self.node, future, timeout_sec=1.0)
-            response = future.result()
-            if response is None:
-                self.node.get_logger().error('/submit_task srv call failed')
-            elif not response.success:
-                self.node.get_logger().error(
-                    'Dispatcher node failed to accept task')
-            else:
-                self.node.get_logger().info(
-                    'Request was successfully submitted '
-                    f'and assigned task_id: [{response.task_id}]')
-        except Exception as e:
-            self.node.get_logger().error('Error! Submit Srv failed %r' % (e,))
+            self.set_parameters([param])
+
+        # Construct task
+        msg = ApiRequest()
+        msg.request_id = "delivery_" + str(uuid.uuid4())
+        payload = {}
+        if self.args.fleet and self.args.robot:
+            self.get_logger().info("Using 'robot_task_request'")
+            payload["type"] = "robot_task_request"
+            payload["robot"] = self.args.robot
+            payload["fleet"] = self.args.fleet
+        else:
+            self.get_logger().info("Using 'dispatch_task_request'")
+            payload["type"] = "dispatch_task_request"
+        request = {}
+
+        # Set task request start time
+        now = self.get_clock().now().to_msg()
+        now.sec = now.sec + self.args.start_time
+        start_time = now.sec * 1000 + round(now.nanosec/10**6)
+        request["unix_millis_earliest_start_time"] = start_time
+
+        def __create_workcell_desc(workcell_list):
+            wl = workcell_list.split(',')
+            return {
+                    "place": wl[0],
```

Comment:
This will throw a runtime error if the user runs the script with fewer comma separated parameters that what this is expecting.